### PR TITLE
🎄 Assortment of smol fixes: f-strings galore, six is zero, pylint still a mess

### LINKED ISSUE #1507: Remove dependence on the six package
#1484 is removing Python 2 support and ultimately this means we can stop using `six`. This issue just allows us to mark-up places that will be affected when we make this change.

### PR DESCRIPTION
**Pull request by nmnobre**
Closes #1507.

### CODE REVIEW

File: src/psyclone/parse/kernel.py [hunk 1]
```diff
@@ -367,52 +363,47 @@ def get_stencil(metadata, valid_types):
...
         raise ParseError(
-            "Internal error, expecting either FunctionVar or "
-            "str from the expression analyser but found {0}".
-            format(type(metadata.args[0])))
+            f"Internal error, expecting either FunctionVar or str from the "
+            f"expression analyser but found {type(metadata.args[0])}")
     if metadata.args[0].args:
         raise ParseError(
             "Expected format stencil(<type>[,<extent>]). However, the "
             "specified <type> '{0}' includes brackets")
```
Comment (nmnobre) [inline src/psyclone/parse/kernel.py:388]: The argument the f-string is missing, maybe it is meant to be `metadata.args[0]`?

File: src/psyclone/tests/psyir/transformations/hoist_trans_test.py [hunk 2]
```diff
@@ -153,12 +153,12 @@ def test_validate_error_read_and_write(fortran_reader, assignment_str):
...
     '''
     psyir = fortran_reader.psyir_from_source(
-        '''subroutine test()
+        f'''subroutine test()
               integer :: i, j, a, b(10)
               do i=1, 10
-                  {0}
+                  {assignment_str}
               enddo
-              end subroutine test'''.format(assignment_str))
```
Comment (nmnobre) [inline src/psyclone/tests/psyir/transformations/hoist_trans_test.py:161]: I kept the indentation as it was, but I can't help but feel it's a bit weird, e.g. `end subroutine` has the same indentation has the subroutine's body....

File: src/psyclone/parse/kernel.py (hunk 1 above)
Comment (sergisiso) [reply src/psyclone/parse/kernel.py:388]: This is going to be deleted soon, so we can ignore it.

File: src/psyclone/tests/psyir/transformations/hoist_trans_test.py (hunk 2 above)
Comment (sergisiso) [reply src/psyclone/tests/psyir/transformations/hoist_trans_test.py:161]: Fixed it myself for now (unless I find bigger issues to request changes)

**Comment by nmnobre** [comment]
@sergisiso You'll have noticed I didn't add myself as an author or as a modifier to any of the files I touched for the first time. We've discussed this before, I'd rather remove all authorship messages than add myself, what do you think for this particular case since I didn't change any semantics?

**Review by sergisiso — APPROVED** [review]
@nmnobre Approved again. The author and dates have been updated and the pycodestyle still clean. The changelog has also been corrected after bringing it up to date with master.

**Pull request merged**